### ISSUE TITLE
Accessibility: Feature page sidebar menu identifiable as a navigation landmark

### ISSUE THREAD

**Issue by sarah-cul** [projectblacklight/spotlight#3582]
### Issue Description - WCAG standard
The sidebar section on feature pages is a navigation menu and should be identified that way to help screen readers better interpret its purpose

### Method of discovery
Use a browser extension such as 'SkipTo Landmarks & Headings' that can list the landmarks and headings for any page

### Steps to Reproduce
View a feature page for an exhibit. Check that the sidebar is identified in a meaningful way. Currently it shows the links as headers (h2). The section is showing as a complimentary section identified as "Limit your search". 

<img width="300"   alt="Image" src="https://github.com/user-attachments/assets/c0a59b91-c420-431f-9810-43464f6eb22f" />

<img width="450"  alt="Image" src="https://github.com/user-attachments/assets/b17176ec-9e23-4659-9e22-bb6637a25c2f" />

### Suggested solution

use `<nav>` tags to wrap the menu, and label it something like "Secondary" or "Feature pages" to distinguish it from the other navigations (main, breadcrumbs, footer).  

<img width="400" height="109" alt="Image" src="https://github.com/user-attachments/assets/3293ee92-9e92-467a-9a03-921a59530b24" />
